addons/website/models/website.py

```diff
@@ -814,7 +814,7 @@ def _website_url(self, cr, uid, ids, field_name, arg, context=None):
         return dict.fromkeys(ids, '#')
 
     def website_publish_button(self, cr, uid, ids, context=None):
-        if self.pool['res.users'].has_group(cr, uid, 'base.group_website_publisher'):
+        if self.pool['res.users'].has_group(cr, uid, 'base.group_website_publisher') and self.open_website_url(cr, uid, ids, context)['url'] != '#':
```

Comment:
Another solution is too add publish buttons in frontend and related urls to all elements which don't have any url, but not a good idea to me. This solution is generic, and @antonylesuisse was ok when I showed it to him.